x/dex/keeper/multihop_swap.go

```diff
@@ -98,69 +101,83 @@ func (k Keeper) RunMultihopRoute(
 	initialInCoin sdk.Coin,
 	exitLimitPrice math_utils.PrecDec,
 	stepCache map[multihopCacheKey]StepResult,
-) (sdk.Coin, func(), error) {
+) (sdk.Coins, sdk.Coin, func(), error) {
 	routeData, err := k.HopsToRouteData(ctx, route.Hops)
 	if err != nil {
-		return sdk.Coin{}, nil, err
+		return sdk.Coins{}, sdk.Coin{}, nil, err
 	}
 	currentPrice := math_utils.OnePrecDec()
 
 	var currentOutCoin sdk.Coin
+	var currentDust sdk.Coin
```

Comment:
A small suggestion, rename all `current` to `step`, i think the `step` name  fits better for variables usage.